src/registry/routes/helpers/get-component.js

```diff
@@ -149,7 +150,12 @@ module.exports = function(conf, repository){
           renderMode: renderMode
         });
 
-        if(isUnrendered){
+        if (skipRendering) {
+          callback({
+            status: 200,
+            response: response
```

Comment:
Agree with you -- will move it around line 78 to skip most of the stuff done further in the file.